src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs

```diff
@@ -1815,14 +1815,14 @@ private static SyntaxTokenList AsModifierList(Accessibility accessibility, Decla
             }
 
             // partial and ref must be last
-            if (modifiers.IsPartial)
+            if (modifiers.IsRef)
             {
-                list = list.Add(SyntaxFactory.Token(SyntaxKind.PartialKeyword));
+                list = list.Add(SyntaxFactory.Token(SyntaxKind.RefKeyword));
             }
 
-            if (modifiers.IsRef)
```

Comment:
A `ref` keyword should be before `partial`, because `ref partial` is only allowed for now. It shouldn't be a breaking change, but anyway pls take a look at this.

Reply:
Look like there was a test gap, thank you for fixing/adding test for this. Could you please also add a test for `readonly ref partial struct` (since the order matters)?

---
In reply to: [249267149](https://github.com/dotnet/roslyn/pull/32037#discussion_r249267149) [](ancestors = 249267149)

Reply:
Done!